### PR TITLE
feat: transaction & blocks timeline & detail views

### PR DESCRIPTION
**Pull request by sunnygleason**
Problem:

* current blocks and transaction views are just mockups

Solution:

* implementation of blocks and transaction views

### CODE REVIEW

File: api/loaders/timeline.js [hunk 1]
```diff
@@ -80,11 +80,39 @@ export async function getTimelinePage(
 ) {
   let timelineKey = `!__timeline:${timeline}`;
 
-  let rawResults = await (direction === '+'
```
Comment (sunnygleason) [inline api/loaders/timeline.js]: we used to just retrieve N + 1 results to detect the "next" conditon

File: api/loaders/timeline.js [hunk 2]
```diff
@@ -80,11 +80,39 @@ export async function getTimelinePage(
...
-    ? redisX.xrangeAsync(timelineKey, start, '+', 'COUNT', count + 1)
-    : redisX.xrevrangeAsync(timelineKey, start, '-', 'COUNT', count + 1));
+  const thePromise = new Promise((resolve, reject) => {
+    try {
+      let tx = redisX.client.multi();
+
+      if (direction === '+') {
```
Comment (sunnygleason) [inline api/loaders/timeline.js]: now, we retrieve N+1 forward and N+1 backward to get "next" and "prev" tokens

File: api/loaders/timeline.js [hunk 3]
```diff
@@ -27,9 +27,7 @@ export async function getTimelineInfo(redisX, timeline, parseValue) {
     try {
       redisX.client
         .multi()
-        .xrange(timelineKey, '-', '+', 'COUNT', 1)
-        .xrevrange(timelineKey, '+', '-', 'COUNT', 1)
-        .xlen(timelineKey)
+        .xinfo('STREAM', timelineKey)
```
Comment (sunnygleason) [inline api/loaders/timeline.js]: one optimized operation should be better than 3

File: api/loaders/transactions/index.js [hunk 4]
```diff
@@ -1,6 +1,35 @@
...
 import {getTimelineInfo, getTimelinePage} from '../timeline';
 import {parseTransaction} from '../parseMessage';
 
+/** see inbound-stream.js#process() -> (message.t === 'entry') */
+async function loadEntryInfo(redisX, entryId) {
+  return await redisX.hgetallAsync([`!ent:${entryId}`]);
+}
+
+// augment with block_id field since it's not available in the timeline
+async function fixupTimelinePage(redisX, timelinePage) {
```
Comment (sunnygleason) [inline api/loaders/transactions/index.js]: `block_id` is a required UI element so we need to fixup transactions with that data

File: src/AppV2.js [hunk 5]
```diff
@@ -90,8 +90,14 @@ const App = () => {
               <Route exact path="/validators/:id" component={ValidatorDetail} />
               <Route exact path="/tour-de-sol" component={TourDeSol} />
               <Route exact path="/blocks" component={Blocks} />
+              <Route exact path="/blocks/timeline/:start" component={Blocks} />
```
Comment (sunnygleason) [inline src/AppV2.js]: add 'timeline' route for pagination

File: src/AppV2.js [hunk 6]
```diff
@@ -90,8 +90,14 @@ const App = () => {
...
               <Route exact path="/blocks/:id" component={BlockDetail} />
               <Route exact path="/transactions" component={Transactions} />
```
Comment (sunnygleason) [inline src/AppV2.js]: add 'timeline' route for pagination

File: src/v2/stores/transactions/timeline.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import {action, flow, observable, decorate} from 'mobx';
```
Comment (sunnygleason) [inline src/v2/stores/transactions/timeline.js]: blocks timeline store is very similar to transactions timeline store

File: src/v2/stores/blocks/timeline.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import {action, flow, observable, decorate} from 'mobx';
```
Comment (sunnygleason) [inline src/v2/stores/blocks/timeline.js]: blocks timeline store is very similar to transactions timeline store

File: src/v2/stores/transactions/detail.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import {action, flow, observable, decorate} from 'mobx';
```
Comment (sunnygleason) [inline src/v2/stores/transactions/detail.js]: blocks detail store is very similar to transactions detail store

File: src/v2/stores/blocks/detail.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import {action, flow, observable, decorate} from 'mobx';
```
Comment (sunnygleason) [inline src/v2/stores/blocks/detail.js]: blocks detail store is very similar to transactions detail store

File: api/loaders/transactions/index.js [hunk 11]
```diff
@@ -17,7 +46,7 @@ export async function loadTransactionIndex(redisX, start, count, direction) {
     parseTransaction,
   );
 
-  const timelinePage = await getTimelinePage(
+  const timelinePageNoPageIds = await getTimelinePage(
```
Comment (CriesofCarrots) [inline api/loaders/transactions/index.js]: Should this be `timelinePageNoBlockIds`?

**Comment by sunnygleason** [comment]
@CriesofCarrots it's all set! Sorry didn't realize you were going through!

File: src/v2/components/Blocks/Table/index.jsx [hunk 12]
```diff
@@ -42,37 +47,84 @@ const tHeads: TableHeadProps[] = [
...
+          <li>
+            <div className={classes.cardTitle}>Time</div>
+            <div title={block.timestamp}>{asTime(block.timestamp)}</div>
+          </li>
+          <li>
+            <div className={classes.cardTitle}>Transactions</div>
+            <div>TODO</div>
+          </li>
+          <li>
+            <div className={classes.cardTitle}>Uncles</div>
```
Comment (CriesofCarrots) [inline src/v2/components/Blocks/Table/index.jsx]: See above, re: confidence

File: src/v2/components/Transactions/Detail/index.jsx [hunk 13]
```diff
@@ -61,7 +85,25 @@ const TransactionDetail = () => {
...
+    },
+    {},
+  );
+
+  const programs = _.values(programMap);
+
+  const tabNav = [
+    `Applications: ${programs.length}`,
+    'status: success',
+    'code/course',
```
Comment (CriesofCarrots) [inline src/v2/components/Transactions/Detail/index.jsx]: "course" > "source"

Comment (sunnygleason) [reply src/v2/components/Transactions/Detail/index.jsx]: I was wondering what that was supposed to be!

Comment (CriesofCarrots) [reply src/v2/components/Transactions/Detail/index.jsx]: It's a lovely little anagram!

**Pull request merged**